### DISCUSSION TITLE
Question about how to extract the exact place that the specific person would go (or something synonym verb like visit, meet, travel)

### DISCUSSION THREAD

**Discussion by icmpnorequest** [explosion/spaCy#12502] [Help: Coding & Implementations] [answered]
Hi,

I tried to extract the place that a specific person would go from a document. I could extract the location entity with Entity Recognition, but couldn't distinguish whether it is the exact place that she/he would go or just mentioned in the text. I was wondering how I could do that?

My opinion is as following:
- Step1: Person entity recognition to recognize the specific person;
- Step2: Dependency parser?

**Comment by svlandeg** [comment]
Hi! Obviously I don't know the specifics of your data and how the sentences look like, but maybe you could try out the [dependencymatcher](https://spacy.io/api/dependencyparser)?

**Comment by DuyguA** [comment, marked as answer]
Hello, if I understood it correctly you'd like to find the person names, right? Sth like this:

```
Jonas visited Italy 2 years ago.
```
In this case, you located the location entity (Italy), then you find the verb that this token attaches (visited). If the verb is one of `go, visit, travel` etc, then, you like to find who made this action. At this point you look for the `nsubj` of the verb i.e. the nominal subject , here Jonas. 

![jonas1](https://user-images.githubusercontent.com/8277232/232316654-24b5f111-cbbd-4c72-a72e-1c3018dedf29.png)

Of course this is a simple sentence and relatively easy case, for more complicated sentences you need more tricks :wink: 


I have a tutorial video for finding subjects of verbs if you want you can check it out :wink:  The playlist is called "spaCy quickies", the video is called [Differentiating Money Entities with Syntax](https://www.youtube.com/watch?v=WY_8vSgkhwA&list=PLJTHlIwB8VcpYWrvV738Yii1V2KquVn1y&index=6) . Cheers!
